### PR TITLE
Chart: More love!

### PR DESCRIPTION
**Pull request by Gacko**

### CODE REVIEW

File: helm/event-exporter-app/templates/configmap.yaml [hunk 1]
```diff
@@ -1,183 +1,176 @@
...
 kind: ConfigMap
 metadata:
-  name: event-exporter-app
-  namespace: giantswarm
+  name: {{ include "event-exporter.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels: {{ include "event-exporter.labels" . | nindent 4 }}
 data:
   config.yaml: |
-    logLevel: debug
```
Comment (AverageMarcus) [inline helm/event-exporter-app/templates/configmap.yaml:8]: Do we no longer want this?

File: helm/event-exporter-app/templates/deployment.yaml [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+  name: {{ include "event-exporter.fullname" . }}
   namespace: {{ .Release.Namespace }}
-  labels:
-    {{- include "labels.common" . | nindent 4 }}
+  labels: {{ include "event-exporter.labels" . | nindent 4 }}
 spec:
+  selector:
+    matchLabels:
+      app: event-exporter-app
+      version: v1
```
Comment (AverageMarcus) [inline helm/event-exporter-app/templates/deployment.yaml:11]: Why not use the `event-exporter.selectorLabels` helper? Why are these different?

File: helm/event-exporter-app/templates/deployment.yaml [hunk 3]
```diff
@@ -15,29 +18,24 @@ spec:
...
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.securityContext }}
+        securityContext: {{ toYaml .Values.securityContext | nindent 10 }}
+        {{- end }}
+        args:
+        - --conf
+        - /etc/event-exporter/config.yaml
+        volumeMounts:
+        - name: config
+          mountPath: /etc/event-exporter
```
Comment (AverageMarcus) [inline helm/event-exporter-app/templates/deployment.yaml:37]: Why the change of the mount path? Is this directory unused within the container image?

File: helm/event-exporter-app/templates/pss-exceptions.yaml [hunk 4]
```diff
@@ -1,28 +0,0 @@
-{{- if .Capabilities.APIVersions.Has "kyverno.io/v2beta1/PolicyException" -}}
```
Comment (AverageMarcus) [inline helm/event-exporter-app/templates/pss-exceptions.yaml:1]: Why is this no longer needed?

File: helm/event-exporter-app/templates/vpa.yaml [hunk 5]
```diff
@@ -1,23 +1,20 @@
-{{ if .Capabilities.APIVersions.Has "autoscaling.k8s.io/v1" }}
```
Comment (AverageMarcus) [inline helm/event-exporter-app/templates/vpa.yaml:1]: Why remove this check?

File: helm/event-exporter-app/Chart.yaml [hunk 6]
```diff
@@ -1,10 +1,10 @@
-apiVersion: v1
+apiVersion: v2
 name: event-exporter-app
-version: [[ .Version ]]
-appVersion: v0.10
-description: The event-exporter-app runs on Control Planes to watch and act upon Kubernetes events.
-home: https://github.com/opsgenie/kubernetes-event-exporter
+description: The Event Exporter runs on management clusters to watch and act upon Kubernetes events.
+version: 0.14.0
+appVersion: "0.9"
```
[the remark this answers is not in the document]
Comment (Gacko) [reply helm/event-exporter-app/Chart.yaml:5]: Yes, because it was wrongly set in the `Chart.yaml` but never bumped in the `values.yaml`, which is being used in the Deployment. I still need to bump it in the `values.yaml` and `Chart.yaml`, but in a different PR.

File: helm/event-exporter-app/templates/configmap.yaml (hunk 1 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/configmap.yaml:8]: Depends on. I'd not use log level debug in production. If we ever need it, we can still add it for actual debugging. But until then it probably only wastes resources, assuming we're shipping logs elsewhere.

File: helm/event-exporter-app/templates/deployment.yaml (hunk 2 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/deployment.yaml:11]: Because you cannot change the selector match labels of an existing Deployment. They are immutable. And since someone decided to just hard-code to this, we now need to keep them.

Only thing I can do is extending the pod template's labels and then use the `event-exporter.selectorLabels` in the NetworkPolicy. But sadly not in the Deployment, at least not as long as I don't remove the whole app from all MCs and re-create it. Like... removing the app from the app collection and adding it again.

File: helm/event-exporter-app/templates/deployment.yaml (hunk 3 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/deployment.yaml:37]: No special reason, just thought it would be better to place config in a directory which actually is about configuration. 😬

File: helm/event-exporter-app/templates/pss-exceptions.yaml (hunk 4 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/pss-exceptions.yaml:1]: Because we had exception on stuff I already fixed by making the security context more concrete and secure. The PSS exceptions would ignore `require-run-as-nonroot`, but as I explicitly set `runAsNonRoot: true`, we no longer need to ignore it. And so we can also just remove all exceptions.

File: helm/event-exporter-app/templates/deployment.yaml (hunk 3 above)
Comment (AverageMarcus) [reply helm/event-exporter-app/templates/deployment.yaml:38]: I'd rather we didn't change the path, just in case something expected it to be there.

File: helm/event-exporter-app/templates/vpa.yaml (hunk 5 above)
Comment (AverageMarcus) [reply helm/event-exporter-app/templates/vpa.yaml:1]: But doesn't that then cause issues with dependencies on creating a new cluster? I recall something like that but not the specifics.

File: helm/event-exporter-app/templates/deployment.yaml (hunk 2 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/deployment.yaml:11]: You know what? You were right! I took another look into the code and we could use the selector labels. I just changed this, new commit incoming.

File: helm/event-exporter-app/templates/deployment.yaml (hunk 3 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/deployment.yaml:38]: Sure, gonna revert that!

File: helm/event-exporter-app/templates/deployment.yaml (hunk 2 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/deployment.yaml]: I updated the helper. Before we were already using the chart name, not the release name for `app:`. I mean, ok, it's still not perfect, but at least it's the same everywhere now.

File: helm/event-exporter-app/templates/deployment.yaml (hunk 3 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/deployment.yaml]: Done.

File: helm/event-exporter-app/templates/vpa.yaml (hunk 5 above)
Comment (Gacko) [reply helm/event-exporter-app/templates/vpa.yaml:1]: Hehe, fun fact: If you add something like this, you will end up with an app not having VPA at all and still being `deployed` successfully. Also not good. With this requirement, the chart operator / Flux Helm controller is just retrying installation until it works.

**Review by AverageMarcus — APPROVED** [review]
